library/src/main/java/com/yayandroid/locationmanager/providers/locationprovider/GooglePlayServicesLocationSource.java

```diff
@@ -89,17 +89,23 @@ void startSettingsApiResolutionForResult(Status status, Activity activity) throw
 
     @SuppressWarnings("ResourceType")
     void requestLocationUpdate() {
-        LocationServices.FusedLocationApi.requestLocationUpdates(googleApiClient, locationRequest, this);
+        if (isGoogleApiClientConnected()) {
+            LocationServices.FusedLocationApi.requestLocationUpdates(googleApiClient, locationRequest, this);
+        } else {
+            connectGoogleApiClient();
```

Comment:
Even if it is, this wouldn't work because `connectGoogleApiClient` doesn't have an immediate action, it will trigger some callbacks which are not defined here. And you'll introduce loop here.